src/contexts/login-context/login-utils.js

```diff
@@ -0,0 +1,130 @@
+      /^(([^<>()[\]\.,;:\s@\"]+(\.[^<>()[\]\.,;:\s@\"]+)*)|(\".+\"))@(([^<>()[\]\.,;:\s@\"]+\.)+[^<>()[\]\.,;:\s@\"]{2,})$/i;
+    return regex.test(email) ? true : false;
+  };
+
+  const isPhoneValid = phone => {
+    return phone.length === 10 ;
+  };
+
+  
+  const validateUser = (mobile,email) => {
+    if (isPhoneValid(mobile) && !isEmailValid(email)) {
+      return {
+        email: "Enter a valid email id",
+        phone: "",
+      };
+    } else if (!isPhoneValid(mobile) && isEmailValid(email)) {
+      return {
+        email: "",
+        phone: "Enter a valid mobile number",
+      };
+    } else if (!isPhoneValid(mobile) && !isEmailValid(email)) {
+      return {
+        email: "Enter a valid email id",
+        phone: "Enter a valid mobile number",
+      };
+    }
+    return {
+      phone: "",
+      email: "",
+    };
+  };
+
+  export {isEmailValid,isPhoneValid,validateUser}
+
+const createUser = async (signupForm,setUserErr,setSignupForm,dispatchUser,navigate) => {
+    const userErrors = validateUser(signupForm.mobile, signupForm.email);
+    setUserErr(userErrors);
+    if (userErrors["phone"] || userErrors["email"]) {
+      return;
+    }
+    try {
+      const response = await axios.post("api/auth/signup", signupForm);
+      if (response.status === 201) {
+        setSignupForm({
+          firstname: "",
+          lastname: "",
+          email: "",
+          password: "",
+          mobile: "",
+        });
+        setUserErr({ email: "", phone: "" });
+        dispatchUser({
+          type: "SIGNUP_SUCCESS",
+          payload: response.data.createdUser,
+        });
+        localStorage.setItem("userToken", response.data.encodedToken);
+        navigate("/");
+        Toast({
+        message: "Sign up successful.",
+        type: "success",
+      });
+      }
+      else{
+        Toast({
+        message: "Signup failed. Please try again.",
+        type: "error",
+      });
+        console.log("signup failed with HTTP status",response.status)
+      }
+    } catch (err) {
+      console.log(err)
+      if (err.response && err.response.status === 422) {
+        Toast({
+        message: "Email already exists, use different one.",
+        type: "error",
+      });
+      }
+      else
+      Toast({
+        message: "Signup failed. Please try again.",
+        type: "error",
+      });
+    }
+  };
+
+const loginUser = async (loginForm,dispatchUser,navigate) => {
+    try {
+      const response = await axios.post("api/auth/login", loginForm);
+      if (response.status === 200) {``
+        localStorage.setItem("userToken", response.data.encodedToken);
+        dispatchUser({ type: "SET_USER_LOGIN", payload: response.data.foundUser });
+        navigate("/");
+        Toast({
+        message: "Signed in successfully.",
+        type: "success",
+      });
+      } else {
+        Toast({
+        message: "Invalid credentials. Please try again.",
+        type: "error",
+      });     
```

Comment:
The Catch would handle this anyway if some error occurs

Reply:
Sometime the status in try could be other than 200 so to handle that I have added this.